crates/biome_js_analyze/src/lint/nursery/use_google_font_preconnect.rs

```diff
@@ -0,0 +1,141 @@
+    ///
+    /// When using Google Fonts, adding the `rel="preconnect"` attribute to the `<link>` tag
+    /// that points to `https://fonts.gstatic.com` is recommended to initiate an early
+    /// connection to the font's origin. This improves page load performance by reducing latency.
+    ///
+    /// Failing to use `preconnect` may result in slower font loading times, affecting user experience.
+    ///
+    /// Note: Next.js automatically adds this preconnect link starting from version 12.0.1, but in cases
+    /// where it's manually added, this rule ensures the `preconnect` attribute is properly used.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.gstatic.com"/>
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link rel="preload" href="https://fonts.gstatic.com"/>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <link rel="preconnect" href="https://fonts.gstatic.com"/>
+    /// ```
+    ///
+    /// ```jsx
+    /// <link href="/logo.svg" rel="icon" />
+    /// ```
+    ///
+    pub UseGoogleFontPreconnect {
+        version: "next",
+        name: "useGoogleFontPreconnect",
+        language: "jsx",
+        sources: &[RuleSource::EslintNext("google-font-preconnect")],
+        source_kind: RuleSourceKind::SameLogic,
+        recommended: false,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+impl Rule for UseGoogleFontPreconnect {
+    type Query = Ast<AnyJsxElement>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        if node.name().ok()?.name_value_token()?.text_trimmed() != "link" {
+            return None;
+        }
+
+        let href = node.get_attribute_inner_string_text("href")?;
+        let rel = node.get_attribute_inner_string_text("rel");
+
+        if href.starts_with("https://fonts.gstatic.com") && (rel.is_none() || rel? != "preconnect")
+        {
+            return Some(node.syntax().text_range());
+        }
+
+        None
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, range: &Self::State) -> Option<RuleDiagnostic> {
+        return Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                range,
+                markup! {
+                    "Attribute "<Emphasis>"rel=\"preconnect\""</Emphasis>" is missing from Google Font link."
+                },
+            )
+            .note(markup!{
+                "Not using "<Emphasis>"preconnect"</Emphasis>" can cause slower font loading and increase latency."
+            })
+        );
+    }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+
+        if node.find_attribute_by_name("rel").is_some() {
+            return None;
+        }
```

Comment:
Is this check needed? The check was already done in `run`. Did I miss something?

Reply:
Yes, I didn't want to change the value if a `rel` already exists, maybe we could do so, I'm not sure :)

The action is a bonus since the original rule doesn't have one.